pkg/genericapiserver/registry/generic/registry/store.go

```diff
@@ -50,94 +50,115 @@ import (
 // object.
 type ObjectFunc func(obj runtime.Object) error
 
-// Store implements pkg/api/rest.StandardStorage.
-// It's intended to be embeddable, so that you can implement any
-// non-generic functions if needed.
-// You must supply a value for every field below before use; these are
-// left public as it's meant to be overridable if need be.
-// This object is intended to be copyable so that it can be used in
-// different ways but share the same underlying behavior.
+// Store implements pkg/api/rest.StandardStorage. It's intended to be
+// embeddable and allows the consumer to implement any non-generic functions
+// that are required. This object is intended to be copyable so that it can be
+// used in different ways but share the same underlying behavior.
+//
+// All fields are required unless specified.
 //
 // The intended use of this type is embedding within a Kind specific
-// RESTStorage implementation. This type provides CRUD semantics on
-// a Kubelike resource, handling details like conflict detection with
-// ResourceVersion and semantics. The RESTCreateStrategy and
-// RESTUpdateStrategy are generic across all backends, and encapsulate
-// logic specific to the API.
+// RESTStorage implementation. This type provides CRUD semantics on a Kubelike
+// resource, handling details like conflict detection with ResourceVersion and
+// semantics. The RESTCreateStrategy, RESTUpdateStrategy, and
+// RESTDeleteStrategy are generic across all backends, and encapsulate logic
+// specific to the API.
 //
 // TODO: make the default exposed methods exactly match a generic RESTStorage
 type Store struct {
-	// Called to make a new object, should return e.g., &api.Pod{}
+	// NewFunc returns a new object, e.g., &api.Pod{}
 	NewFunc func() runtime.Object
 
-	// Called to make a new listing object, should return e.g., &api.PodList{}
+	// NewListFunc returns a new list object for this resource, e.g., &api.PodList{}.
```

Comment:
Note that it needs to be registered with the scheme and you may want to relate it to the API.  This is the object you return from a list request `curl GET /apis/group/version/namespaces/ns-name/myresource`